### PR TITLE
Fix deletion of already committed pending snapshots

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

FilebasedSnapshotStore keeps track of a list of pending snapshots which will be used to deleted them. When a pending snapshot is committed, it was not deleted from the list. Later when snapshot store purges all pending snapshots, it tries to delete also this non existing pending snapshot which caused unnecessary warning in the log message. This PR fixes it by removing the pending snapshot from the list after a snapshot is successfully committed.

Blocked by #7153 

## Related issues

closes #

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedReceivedSnapshot.java [hunk 1]
```diff
@@ -211,7 +211,7 @@ public Path getPath() {
   private void abortInternal() {
     try {
       LOGGER.debug("Aborting received snapshot in dir {}", directory);
-      FileUtil.deleteFolder(directory);
+      FileUtil.deleteFolderIfExists(directory);
     } catch (final NoSuchFileException nsfe) {
```
Comment (MiguelPires) [inline snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedReceivedSnapshot.java:215]: This catch can be removed

**Review by miguelpires — CHANGES_REQUESTED** [review]
Looks good. I just had one comment

File: snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedReceivedSnapshot.java (hunk 1 above)
Comment (deepthidevaki) [reply snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedReceivedSnapshot.java]: :+1:

**Review by miguelpires — APPROVED** [review]
Thanks :+1:

**Pull request merged**